server/lib/transactions.ts

```diff
@@ -213,6 +215,84 @@ export async function createTransactionsFromPaidExpense(
   return models.Transaction.createDoubleEntry(transaction);
 }
 
+export async function createTransactionsForManuallyPaidExpense(
+  host,
+  expense,
+  paymentProcessorFeeInHostCurrency,
+  totalAmountPaidInHostCurrency,
+  /** Will be stored in transaction.data */
+  transactionData: Record<string, unknown> = {},
+) {
+  assert(paymentProcessorFeeInHostCurrency >= 0, 'Payment processor fee must be positive');
+  assert(totalAmountPaidInHostCurrency >= 0, 'Total amount paid must be positive');
+
+  const isCoveredByPayee = expense.feesPayer === 'PAYEE';
+  const amount = toNegative(totalAmountPaidInHostCurrency - paymentProcessorFeeInHostCurrency);
```

Comment:
As per the comment above (fees are paid by the collective) I believe this should be:

```suggestion
  const amount = toNegative(totalAmountPaidInHostCurrency + paymentProcessorFeeInHostCurrency);
```

Reply:
Since from now on the host admin is informing the total amount paid (this means the net amount that left their bank account, including fees), here we're just separating the real amount related to the expense from the fees.

The logic of who's paying for the fee comes later on.

Reply:
> The logic of who's paying for the fee comes later on.

Where? From what I see, we're jumping straight from this to `Transaction.createDoubleEntry` without checking the `feesPayer`. Don't we need something like https://github.com/opencollective/opencollective-api/blob/3863342448edba0748ea95bd6cf8964c2092c005/server/lib/transactions.ts#L206 to record fees on the right account?

Reply:
I'm sorry @Betree, you're correct. I'll implement this conditional and re-submit for review.